src/core/syntax.ml

```diff
@@ -245,35 +247,42 @@ type p_config =
   | P_config_unif_rule of p_rule
   (** Unification hint declarations. *)
 
-(** Parser-level representation of a single command. *)
-type p_statement = (ident * p_arg list * p_type option) loc
-
 (** Parser-level representation of modifiers. *)
-type p_modifier =
+type p_modifier_aux =
   | P_mstrat of Terms.match_strat (** pattern matching strategy *)
   | P_expo of Terms.expo (** visibility of symbol outside its modules *)
   | P_prop of Terms.prop (** symbol properties : constant, definable, ... *)
-  | P_opaq of Terms.opacity (** opacity of the definition *)
+  | P_opaq (** opacity *)
+
+type p_modifier = p_modifier_aux loc
 
-let is_opaq : p_modifier loc -> bool = fun {elt; _} ->
-  match elt with
-  | P_opaq(Opaque) -> true
-  | _ -> false
+let is_prop {elt; _} = match elt with P_prop(_) -> true | _ -> false
+let is_opaq {elt; _} = match elt with P_opaq -> true | _ -> false
+let is_expo {elt; _} = match elt with P_expo(_) -> true | _ -> false
+let is_mstrat {elt; _} = match elt with P_mstrat(_) -> true | _ -> false
 
+type p_symbol =
+  { p_sym_mod : p_modifier list (** modifiers *)
+  ; p_sym_nam : ident (** symbol name *)
+  ; p_sym_arg : p_arg list (** arguments before ":" *)
+  ; p_sym_typ : p_type option (** symbol type *)
+  ; p_sym_trm : p_term option (** symbol definition *)
+  ; p_sym_prf : (p_tactic list * p_proof_end) option (** proof script *)
+  ; p_sym_def : bool (** is the symbol defined ? *) }
+
+(** Parser-level representation of a single command. *)
 type p_command_aux =
   | P_require    of bool * p_module_path list
   (** Require statement (require open if the boolean is true). *)
   | P_require_as of p_module_path * (string * bool) loc
   (** Require as statement. *)
   | P_open       of p_module_path list
   (** Open statement. *)
-  | P_symbol     of p_modifier loc list * p_statement * p_term option
-                    * (p_tactic list * p_proof_end loc) option
-                    * Terms.proof_meaning
+  | P_symbol     of p_symbol
```

Comment:
How to define eq_p_symbol then?

Reply:
Ah that's true. Perhaps providing provide an `eq_p_command` is enough (I'm not sure how it is used)? If it's the case, the change is slightly bigger, but it may be interesting.